ops.qmd

```diff
@@ -2,84 +2,760 @@
+* Monitoring model and infrastructure performance using tools like Prometheus, Grafana, ELK stack. Building alerts and dashboards.
+* Implementing governance practices around model development, testing, and promotion. Enabling reproducibility and traceability. 
+* Embedding ML models within applications. Exposing models via APIs and microservices for integration.
+* Optimizing infrastructure performance and costs. Leveraging autoscaling, spot instances, and availability across regions.
+
+For example, a DevOps engineer provisions a Kubernetes cluster on AWS using Terraform to run ML training jobs and online deployment. They build a CI/CD pipeline in Jenkins which triggers model retraining if new data is available. After automated testing, the model is registered with MLflow and deployed in the Kubernetes cluster. The engineer then monitors cluster health, container resource usage, and API latency using Prometheus and Grafana.
+
+The DevOps team enables rapid experimentation and reliable deployments for ML through expertise in cloud, automation, and monitoring. Their work maximizes model impact while minimizing technical debt.
+
+### Project Managers
+
+Project managers play a vital role in MLOps by coordinating the activities between the different teams involved in delivering ML projects. They help drive alignment, accountability, and accelerated results. Their main responsibilities include:
+
+* Working with stakeholders to define project goals, success metrics, timelines and budgets. Outlining specifications and scope.
+* Creating a project plan spanning activities like data acquisition, model development, infrastructure setup, deployment, and monitoring.
+* Coordinating design, development and testing efforts between data engineers, data scientists, ML engineers and DevOps roles.
+* Tracking progress and milestones. Identifying roadblocks and resolving through corrective actions. Managing risks and issues.
+* Facilitating communication through status reports, meetings, workshops, documentation. Enabling seamless collaboration.
+* Driving adherence to timelines and budget. Escalating anticipated overruns or shortfalls for mitigation.
+
+For example, a project manager would create a project plan for the development and ongoing enhancement of a customer churn prediction model. They coordinate between data engineers building data pipelines, data scientists experimenting with models, ML engineers productionalizing models, and DevOps setting up deployment infrastructure. The project manager tracks progress via milestones like dataset preparation, model prototyping, deployment, and monitoring. They surface any risks, delays or budget issues to enact preventive solutions.
+
+Skilled project managers enable MLOps teams to work synergistically to deliver maximum business value from ML investments rapidly. Their leadership and organization align with diverse teams.
+
+## Challenges in Embedded MLOps
```

Comment:
I wonder if it'd be better for the "Traditional MLOps vs. Embedded MLOps" section to come before this section? The transition felt a bit abrupt to me...

Reply:
Hmm. The issue is the title I think. so I changed the title to just be a review of the system challenges, and then we get into the Embedded Ops section. Hope that resolves the confusion

Reply:
I agree with this